File: app/src/main/java/com/callscribe/export/CsvExporter.kt
package com.callscribe.export

import android.content.Context
import com.callscribe.R
import com.callscribe.data.TaskRow
import com.callscribe.ui.Format
import java.util.Locale

object CsvExporter {

    private val headerIds = intArrayOf(
        R.string.csv_created, R.string.csv_source, R.string.csv_contact, R.string.csv_phone,
        R.string.csv_task, R.string.csv_details, R.string.csv_due, R.string.csv_all_day,
        R.string.csv_priority, R.string.csv_category, R.string.csv_status,
        R.string.csv_confidence, R.string.csv_reminder, R.string.csv_quote
    )

    fun toCsv(context: Context, tasks: List<TaskRow>): String {
        val builder = StringBuilder()
        // BOM, за да отвори Excel кирилицата коректно.
        builder.append('﻿')
        builder.append(headerIds.joinToString(",") { escape(context.getString(it)) }).append("\r\n")

        val yes = context.getString(R.string.yes)
        val no = context.getString(R.string.no)

        for (task in tasks) {
            val row = listOf(
                Format.dateTime(task.createdAt),
                Format.source(context, task.source),
                task.contactName.orEmpty(),
                task.phone.orEmpty(),
                task.title,
                task.details.orEmpty(),
                task.dueAt?.let { Format.due(it, task.allDay) }.orEmpty(),
                if (task.allDay) yes else no,
                Format.priority(context, task.priority),
                task.category.orEmpty(),
                Format.status(context, task.status),
                String.format(Locale.US, "%.2f", task.confidence),
                task.reminderAt?.let { Format.dateTime(it) }.orEmpty(),
                task.quote.orEmpty()
            )
            builder.append(row.joinToString(",") { escape(it) }).append("\r\n")
        }
        return builder.toString()
    }

    private fun escape(value: String): String {
        val cleaned = value.replace("\r\n", " ").replace('\n', ' ').replace('\r', ' ')
        return "\"" + cleaned.replace("\"", "\"\"") + "\""
    }
}

File: ios/App/Sources/Services/CsvExporter.swift
import Foundation

enum CsvExporter {

    private static var headers: [String] {
        [
            String(localized: "Created"), String(localized: "Source"),
            String(localized: "Contact"), String(localized: "Task"),
            String(localized: "Details"), String(localized: "Due"),
            String(localized: "All day"), String(localized: "Priority"),
            String(localized: "Category"), String(localized: "Status"),
            String(localized: "Confidence"), String(localized: "Reminder"),
            String(localized: "Quote")
        ]
    }

    static func csv(for tasks: [TaskRow]) -> String {
        // BOM, за да отвори Excel кирилицата коректно.
        var output = "\u{FEFF}"
        output += headers.map(escape).joined(separator: ",") + "\r\n"

        for task in tasks {
            let row = [
                Formatting.dateTime(task.createdAt),
                Formatting.source(task.source),
                task.contactName ?? "",
                task.title,
                task.details ?? "",
                task.dueAt.map { Formatting.due($0, allDay: task.allDay) } ?? "",
                task.allDay ? String(localized: "yes") : String(localized: "no"),
                Formatting.priority(task.priority),
                task.category ?? "",
                Formatting.status(task.status),
                String(format: "%.2f", task.confidence),
                task.reminderAt.map(Formatting.dateTime) ?? "",
                task.quote ?? ""
            ]
            output += row.map(escape).joined(separator: ",") + "\r\n"
        }
        return output
    }

    /// Записва CSV във временен файл, готов за споделяне.
    static func writeTemporaryFile(for tasks: [TaskRow]) -> URL? {
        let url = FileManager.default.temporaryDirectory
            .appendingPathComponent("napomnyania.csv")
        do {
            try csv(for: tasks).data(using: .utf8)?.write(to: url, options: .atomic)
            return url
        } catch {
            return nil
        }
    }

    private static func escape(_ value: String) -> String {
        let cleaned = value
            .replacingOccurrences(of: "\r\n", with: " ")
            .replacingOccurrences(of: "\n", with: " ")
            .replacingOccurrences(of: "\r", with: " ")
        return "\"" + cleaned.replacingOccurrences(of: "\"", with: "\"\"") + "\""
    }
}
